File: AN_2/SEM_2/MPP/c#/lab4/lab4/domain/Echipa.cs
namespace lab4.domain
{
    public class Echipa(int idEchipa, string nume) : IDentifiable<int>
    {
        private int _idEchipa = idEchipa;
        private string _nume = nume;

        public Echipa(string nume) : this(0, nume) {}

        public void SetId(int id) => this._idEchipa = id;
        public int GetId() => _idEchipa;

        public string GetNume() => _nume;
        public void SetNume(string nume) => this._nume = nume;

        public override bool Equals(object? obj)
        {
            if (this == obj) return true;
            if (obj == null || GetType() != obj.GetType()) return false;
            var echipa = (Echipa)obj;
            return _idEchipa == echipa._idEchipa;
        }

        public override int GetHashCode() => _idEchipa.GetHashCode();

        public override string ToString()
        {
            return $"Echipa{{idEchipa={_idEchipa}, nume='{_nume}'}}";
        }
    }
}

File: AN_2/SEM_2/MPP/java/lab6/Model/src/main/java/ro/mpp2024/model/Echipa.java
package ro.mpp2024.model;

import java.io.Serializable;
import java.util.Objects;

public class Echipa implements Identifiable<Integer>, Serializable {
    private int idEchipa;
    private String nume;

    public Echipa(String nume) {
        this.nume = nume;
    }

    public Echipa(int idEchipa,String nume) {
        this.idEchipa = idEchipa;
        this.nume = nume;
    }

    @Override
    public void setId(Integer integer) {
        this.idEchipa = integer;
    }

    @Override
    public Integer getId() {
        return idEchipa;
    }

    public void setIdEchipa(int idEchipa) {
        this.idEchipa = idEchipa;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Echipa echipa = (Echipa) o;
        return idEchipa == echipa.idEchipa;
    }

    @Override
    public int hashCode() {
        return Objects.hash(idEchipa);
    }

    @Override
    public String toString() {
        return "Echipa{" +
                "idEchipa=" + idEchipa +
                ", nume='" + nume + '\'' +
                '}';
    }

}
